
# This is a patch for BioPerl-1.6.1 to update it to BioPerl-1.6.1-patched
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -c "BioPerl-1.6.1\maintenance\symlink_script.pl" "BioPerl-1.6.1-patched\maintenance\symlink_script.pl"
Index: ./maintenance/symlink_script.pl
*** ./maintenance/symlink_script.pl	Tue Sep 29 10:33:45 2009
--- ./maintenance/symlink_script.pl	Wed Feb 17 07:14:58 2010
***************
*** 18,25 ****
  chdir($install_dir);
  
  while (my ($source, $destination) = each %symlink_scripts) {
!     eval { symlink($source, $destination) };
!     $build->log_warn("Cannot create symbolic link named $destination on your system for $source in $install_dir\n") if $@;
  }
  
  chdir($orig_dir);
--- 18,32 ----
  chdir($install_dir);
  
  while (my ($source, $destination) = each %symlink_scripts) {
! 	if ($^O !~ /Win32/) {
! 		eval { symlink($source, $destination) };
! 		$build->log_warn("Cannot create symbolic link named $destination on your system for $source in $install_dir\n") if $@;
! 	} else {
! 		# Win32 perl does not implement symlink(), as it would not work on all filesystems.
! 		require File::Copy;
! 		eval { File::Copy::copy($source, $destination) };
! 		$build->log_warn("Cannot create copy of script named $destination on your system for $source in $install_dir\n") if $@;
! 	}
  }
  
  chdir($orig_dir);
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Wed Feb 17 07:21:43 2010
# Generated by        : makepatch 2.04
# Recurse directories : Yes
# Excluded files      : (\A|/).*\~\Z
#                       (\A|/).*\.a\Z
#                       (\A|/).*\.bak\Z
#                       (\A|/).*\.BAK\Z
#                       (\A|/).*\.elc\Z
#                       (\A|/).*\.exe\Z
#                       (\A|/).*\.gz\Z
#                       (\A|/).*\.ln\Z
#                       (\A|/).*\.o\Z
#                       (\A|/).*\.obj\Z
#                       (\A|/).*\.olb\Z
#                       (\A|/).*\.old\Z
#                       (\A|/).*\.orig\Z
#                       (\A|/).*\.rej\Z
#                       (\A|/).*\.so\Z
#                       (\A|/).*\.Z\Z
#                       (\A|/)\.del\-.*\Z
#                       (\A|/)\.make\.state\Z
#                       (\A|/)\.nse_depinfo\Z
#                       (\A|/)core\Z
#                       (\A|/)tags\Z
#                       (\A|/)TAGS\Z
# p "maintenance/symlink_script.pl" 2350 1266416098 0100666
#### End of ApplyPatch data ####

#### End of Patch kit [created: Wed Feb 17 07:21:43 2010] ####
#### Patch checksum: 64 2512 40353 ####
#### Checksum: 82 3207 32276 ####
